src/required_resources.cpp

```diff
@@ -93,36 +93,27 @@ int RequiredResources::create_maps_impl(BPFtrace &bpftrace, bool fake)
   }
   if (needs_data_map)
   {
-    // get size of all the format strings
-    size_t size = 0;
-    for (auto &it : seq_printf_args)
-      size += std::get<0>(it).size() + 1;
-
-    // compute buffer size to hold all the formats and create map with that
-    int ptr_size = sizeof(unsigned long);
-    size = (size / ptr_size + 1) * ptr_size;
-    auto map = std::make_unique<T>(
-        "data", libbpf::BPF_MAP_TYPE_ARRAY, 4, size, 1, 0);
-
-    // copy all the format strings to buffer, head to tail
-    uint32_t idx = 0;
-    std::vector<uint8_t> formats(size, 0);
-    for (auto &arg : seq_printf_args)
+    auto map = prepareFormatStringDataMap<T>(seq_printf_args, fake);
+    if (map == nullptr || is_invalid_map(map->mapfd_))
     {
-      auto str = std::get<0>(arg).c_str();
-      auto len = std::get<0>(arg).size();
-      memcpy(&formats.data()[idx], str, len);
-      idx += len + 1;
+      failed_maps += 1;
     }
-
-    // store the data in map
-    uint64_t id = 0;
-    int ret = bpf_update_elem(map->mapfd_, &id, formats.data(), 0);
-
-    if (is_invalid_map(map->mapfd_) || (!fake && ret == -1))
+    else
```

Comment:
I'm not familiar enough with this code to know why it's like this myself, but why doesn't this new code follow the style in the other branches of always calling `bpftrace.maps.Set()`?

If this way is better then it'd be best to change them all - the mismatch in styles adds extra confusion (to me at least!)